strum_macros/src/enum_iter.rs

```diff
@@ -99,10 +107,24 @@ pub fn enum_iter_inner(ast: &syn::DeriveInput) -> TokenStream {
             }
         }
 
+        impl #impl_generics DoubleEndedIterator for #iter_name #ty_generics #where_clause {
+            fn next_back(&mut self) -> Option<#name #ty_generics> {
+                if self.back_idx >= #variant_count {
```

Comment:
Same thing here, the iterators shouldn't overlap so if `back_idx` is less than `idx`, we are done.